battle.cpp

```diff
@@ -119,18 +119,8 @@ void WriteStats(Battle &battle, Army &army, StatsCategory category) {
 	}
 }
 
-Battle::Battle()
-{
-	asstext = 0;
-}
-
-Battle::~Battle()
-{
-	if (asstext)
-	{
-		delete asstext;
-	}
-}
+Battle::Battle() { }
+Battle::~Battle() { }
```

Comment:
Since I moved asstext (I didn't rename it, but should have!) from a Astring* to a std::string there is no longer any need to free.